### PR TITLE
JavaScript library

### PR DESCRIPTION
**Pull request by mahmoud717**
Pull request to merge js-library branch to development branch.

- Consists in a Library app that allows users to add books to a list and delete it and change the "read" to true or false.

Built With:

- HTML
- CSS
- JavaScript

### CODE REVIEW

**Comment by luciano-ilha** [comment]
Pull request to merge js-library branch to development branch.

- Consists in a Library app that allows users to add books to a list and delete it and change the "read" to true or false.

Built With:

- HTML
- CSS
- JavaScript

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @luciano-ilha and @mahmoud717 
### You have done really well so far :clap: However, you have a few issues to fix:-
- [x] We recommend you use es6 arrow functions to implement your code base, see below example
```
const showBook = (title, author, pages, read) => {
  # your code
}
```
- [x] Install all the required dependencies, configure them properly and run command for linting, if errors kindly fix them. [Follow this guide](https://github.com/microverseinc/linters-config/tree/master/javascript)
- [x] Separate Dom manipulation from app logic
### Make the changes and submit for another code review. Good luck.
- Happy coding!

File: script.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  li.appendChild(bookRead);
+  li.appendChild(deleteBtn);
+  $bookList.appendChild(li);
+}
+function switchBack() {
+  const hidden = document.querySelector('.hidden-form');
+  hidden.classList.add('hidden');
+  const wrapper = document.querySelector('#wrapper');
+  wrapper.classList.remove('hidden');
+}
```
Comment (rOluochKe) [inline script.js]: ### Separate dom manipulation from app logic

Comment (mahmoud717) [reply script.js]: ### Hi there, regarding the previous TSE's requirements:
---
- we changed all of the functions to arrow functions.
- the previous TSE asked us to configure and correct our linters errors, we already did that.
- he requested from us that we should separate the DOM manipulation syntax, but no function has both together.
---

If you find any other issues, please be more clear about where the issues are.

**Review by rOluochKe — CHANGES_REQUESTED** [review]
### Following the previous review, you have not implemented all the changes;-
- [ ] Convert all the functions to es6 arrow function, you forgot about this `function Book(title, author, pages, read)`
- [ ] Remove the file .DS_Store is not required in the project.
- [ ] Separate the dom manipulation from app logic
### Make the changes and submit for another code review. Good luck.
- Happy coding!

**Comment by mahmoud717** [comment]
function Book(title, author, pages, read) is a constructor fuction 
- and once again, where is the separation neeeeeeeeded..

**Review by rOluochKe — CHANGES_REQUESTED** [review]
### Kindly implement all the changes from the previous reviews.

**Comment by rOluochKe** [comment]
> function Book(title, author, pages, read) is a constructor fuction
> 
> * and once again, where is the separation neeeeeeeeded..

- I commented it earlier, the first review. [check here](https://github.com/mahmoud717/js-library/pull/1/files/6b83cf713dac4c0f643057ac183e3e8b477d8e2a)
- yes, you can change the constructor function to arrow function

**Comment by mahmoud717** [comment]
the project specification doesn't require me separating my code, it was a mistake by the tse.

File: script.js [hunk 2]
```diff
@@ -1,165 +1,165 @@
...
-    this.pages = pages;
-    this.read = read;
+const library = [];
+const $bookList = document.querySelector('ul');
+const $addBookButton = document.querySelector('.add-book-button');
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
```
Comment (lucasmazo23) [inline script.js:10]: - [ ] REQUIRED This function is still using the `function` keyword instead of arrow functions syntax.

**Review by lucasmazo23 — CHANGES_REQUESTED** [review]
## Changes required 🔧
### You are doing a great job so far, and the project is looking good.
 
### For the project to get approved kindly implement these changes:
 
### Required changes:
- [ ] [REQUIRED] There's still a function using the function keyword instead of arrow functions syntax. (check my comment below)

- Regarding the separation of logic and DOM Manipulation, here's a [link ](https://microverse.pathwright.com/library/javascript-module-v10-copy/166786/path/step/97635156/) where they asks you to keep the application logic separated from DOM manipulation tasks.
image


### Happy coding! 💻
## Lucas Mazo

**Comment by mahmoud717** [comment]
```
function Book(title, author, pages, read) {
  this.title = title;
  this.author = author;
  this.pages = pages;
  this.read = read;}
```
this function is a constructor function and cannot be converted into an arrow function.
also, there are no logic code and DOM code mixed in the same function.

please take that in mind when reviewing

**Comment by mahmoud717** [comment]
> ## Changes required 🔧
> ### You are doing a great job so far, and the project is looking good.
> ### For the project to get approved kindly implement these changes:
> ### Required changes:
> * [ ]  [REQUIRED] There's still a function using the function keyword instead of arrow functions syntax. (check my comment below)
> * Regarding the separation of logic and DOM Manipulation, here's a [link ](https://microverse.pathwright.com/library/javascript-module-v10-copy/166786/path/step/97635156/) where they asks you to keep the application logic separated from DOM manipulation tasks.
>   image
> 
> ### Happy coding! 💻
> ## Lucas Mazo
The previous tse meant that i should separate them into two disserent files

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @luciano-ilha and @mahmoud717 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**